pkg/controller/workload/workload_event.go

```diff
@@ -108,19 +108,6 @@ func (svc *ServiceEvent) processWorkloadResponse(rsp *service_discovery_v3.Delta
 			err = fmt.Errorf("unsupport type url %s", rsp.GetTypeUrl())
 		}
 	}
-
-	if rsp.RemovedResources != nil {
```

Comment:
is there a situation we need to consider？ rsp.GetResources() == nil & rsp.RemovedResources != nil

Reply:
I have checked, in the function we use for range, if this is nil, it does not matter. 

Just we need to guarantee all the functions removeXXX or handleXX is safe with nil input

Reply:
we need also change this condition to enter handlexx function， when rsp.GetResources() == nil  but rsp.RemovedResources != nil, should we ? line 101 :
`	if rsp.GetResources() != nil {
		switch rsp.GetTypeUrl() {
		case AddressType:
			err = handleAddressTypeResponse(rsp)
		case AuthorizationType:
			err = svc.handleAuthorizationTypeResponse(rsp)
		default:
			err = fmt.Errorf("unsupport type url %s", rsp.GetTypeUrl())
		}
	}`

Reply:
Not sure i can get what you mean

Reply:
> Not sure i can get what you mean

in processWorkloadResponse function ，The current code implementation misses a situation ：when rsp has RemovedResources field without Resources field